apps/passport-client/components/screens/ImportBackupScreen.tsx

```diff
@@ -0,0 +1,386 @@
+// to import.
+type ImportState =
+  // Initial state
+  | { state: "initial" }
+  // The selected file is valid, and the user can decide whether to import it.
+  | {
+      state: "valid-file-selected";
+      // The collection parsed from the selected file
+      collection: PCDCollection;
+      // The PCD IDs referring to PCDs which are valid to import
+      mergeablePcdIds: Set<PCD["id"]>;
+      // The folders the user has selected in the UI
+      selectedFolders: Set<string | symbol>;
+      // The PCD IDs that are valid and within the selected folders
+      selectedPcdIds: Set<PCD["id"]>;
+      // The count of valid PCDs in each importable folder
+      folderCounts: Record<string | symbol, number>;
+    }
+  // The import has been carried out, and `added` is the number of PCDs
+  // imported.
+  | { state: "import-complete" }
+  // The selected file is not valid.
+  | { state: "invalid-file" };
+
+export function ImportBackupScreen() {
+  const [importState, setImportState] = useState<ImportState>({
+    state: "initial"
+  });
+
+  // Global application state, used to report the success or failure of the
+  // import.
+  const importScreenState = useImportScreenData();
+
+  const { openFilePicker, filesContent } = useFilePicker({
+    accept: ".json",
+    multiple: false,
+    onFilesSelected: () => {
+      setImportState({ state: "initial" });
+    }
+  });
+
+  const existingPcdCollection = usePCDCollection();
+  const dispatch = useDispatch();
+
+  // Called when a valid file has been selected, and the user chooses to import
+  // PCDs from it.
+  const importPCDs = useCallback(() => {
+    // Should never happen, but makes TypeScript happy that we checked for it
+    if (importState.state !== "valid-file-selected") return;
+
+    dispatch({
+      type: "merge-import",
+      collection: importState.collection,
+      pcdsToMergeIds: importState.selectedPcdIds
+    });
+
+    setImportState({ state: "import-complete" });
+  }, [dispatch, importState]);
+
+  // Responds to the user having selected a file to import, or to changes in
+  // the user's current PCD collection.
+  useEffect(() => {
+    (async () => {
+      // If a file has been selected, and isn't invalid or already imported
+      if (
+        filesContent.length > 0 &&
+        importState.state !== "import-complete" &&
+        importState.state !== "invalid-file"
+      ) {
+        let parsedCollection: PCDCollection;
+
+        // If the file hasn't been processed yet, process it
+        if (importState.state === "initial") {
+          try {
+            // Parse the file content as JSON
+            const storageExport = JSON.parse(filesContent[0].content);
+            // Deserialize the storage - throws an error if the content is not
+            // recognized
+            parsedCollection = (
+              await deserializeStorage(storageExport, await getPackages())
+            ).pcds;
```

Comment:
may be worth adding an issue in github for later pointing to this location for importing subscriptions